ycmd/extra_conf_store.py

```diff
@@ -233,5 +234,5 @@ def _RandomName():
 
 
 def _GlobalYcmExtraConfFileLocation():
-  return os.path.expanduser(
-    user_options_store.Value( 'global_ycm_extra_conf' ) )
+  return os.path.expandvars( os.path.expanduser(
```

Comment:
Maybe we should use vim's `expand()` function instead of Python's ?

I wonder if that would be more obvious to users? Just a thought.